kyo-core/shared/src/main/scala/kyo/streams.scala

```diff
@@ -0,0 +1,158 @@
+package kyo
+
+import kyo.*
+import kyo.core.*
+import kyo.core.internal.*
+
+class Streams[V] extends Effect[Streams[V]]:
+    type Command[T] = V
+
+end Streams
+
+object Streams:
+
+    import internal.*
+
+    class Done
+    object Done extends Done
+
+    private case object streams extends Streams[Any]
+    def apply[V]: Streams[V] = streams.asInstanceOf[Streams[V]]
+
+    extension [V](s: Streams[V])
+
+        def emit(v: V): Unit < Streams[V] =
+            suspend(s)(v)
+
+        def emit(v: V, tail: V*): Unit < Streams[V] =
+            emit(v +: tail)
+
+        def emit(v: Seq[V]): Unit < Streams[V] =
+            if v.isEmpty then ()
+            else emit(v.head).andThen(emit(v.tail))
+
+        def buffer[T: Flat](size: Int)(v: T < Streams[V])(
+            using Flat[V]
+        ): T < (Streams[V] & Fibers) =
+            Channels.init[V | Done](size).map { ch =>
+                Fibers.init(runChannel(ch)(v)).map(f => emit(ch).andThen(f.get))
```

Comment:
Ideally, `v` should allow other effects (`T < (Streams[V] & S)`) but it's not possible because of https://github.com/getkyo/kyo/issues/229.